cargo-shuttle/src/lib.rs

```diff
@@ -343,16 +345,32 @@ impl Shuttle {
         Ok(())
     }
 
-    async fn logs(&self, client: &Client, id: Option<Uuid>, follow: bool) -> Result<()> {
+    async fn logs(
+        &self,
+        client: &Client,
+        id: Option<Uuid>,
+        latest: bool,
+        follow: bool,
+    ) -> Result<()> {
         let id = if let Some(id) = id {
             id
         } else {
-            let summary = client.get_service(self.ctx.project_name()).await?;
+            let proj_name = self.ctx.project_name();
+
+            if latest {
+                // Find latest deployment (not always an active one)
+                let deployments = client.get_deployments(proj_name).await?;
+                if deployments.is_empty() {
+                    bail!("Could not find any deployments for '{}'. Try passing a deployment ID manually", proj_name);
+                }
+                let most_recent = deployments.last().unwrap(); // length 0 is checked above
```

Comment:
This is a bit of a nitpick since `get_deployments` will almost always 404 if there are no deployments, because if there are no deployments there is most likely no service (although it is possible).
```suggestion
                let most_recent = deployments.last().context(format!(
                    "Could not find any deployments for '{proj_name}'. Try passing a deployment ID manually",
                ))?;
```